test/net/sourceforge/kolmafia/request/FightRequestTest.java

```diff
@@ -222,26 +222,34 @@ public void handlesIngredientsCharge(String file, int chargeInitial, int chargeE
 
     @ParameterizedTest
     @CsvSource({
-      "test_fight_cookbookbat_quest_new_1.html,false,skullery maid,The Haunted Kitchen",
-      "test_fight_cookbookbat_quest_new_2.html,false,crate,Noob Cave",
-      "test_fight_cookbookbat_quest_new_3.html,false,novelty tropical skeleton,The Skeleton Store",
-      "test_fight_cookbookbat_quest_reminder_1.html,true,,The Haunted Kitchen",
-      "test_fight_cookbookbat_quest_reminder_2.html,true,crate,",
-      "test_fight_cookbookbat_quest_reminder_3.html,true,horrible tourist family,Barf Mountain",
+      "test_fight_cookbookbat_quest_new_1.html,false,skullery maid,The Haunted Kitchen,Vegetable of Jarlsberg",
+      "test_fight_cookbookbat_quest_new_2.html,false,crate,Noob Cave,Yeast of Boris",
+      "test_fight_cookbookbat_quest_new_3.html,false,novelty tropical skeleton,The Skeleton Store,St. Sneaky Pete's Whey",
+      "test_fight_cookbookbat_quest_reminder_1.html,true,,The Haunted Kitchen,",
+      "test_fight_cookbookbat_quest_reminder_2.html,true,crate,,",
+      "test_fight_cookbookbat_quest_reminder_3.html,true,horrible tourist family,Barf Mountain,",
     })
     public void handlesQuest(
-        String file, boolean isReminder, String monsterName, String locationName) {
+        String file,
+        boolean isReminder,
+        String monsterName,
+        String locationName,
+        String ingredientName) {
+      if (monsterName == null) monsterName = "";
+      if (locationName == null) locationName = "";
+      if (ingredientName == null) ingredientName = "";
```

Comment:
These bother me.  The parameters are entirely under your control and if they are null then perhaps you should revisit how they are being set so that you don't have to check and reset.

I'll call this blocking at least until I understand why the fix is not as simple as I think it should be.